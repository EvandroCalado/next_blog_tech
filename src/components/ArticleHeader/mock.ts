export default {
  id: '3',
  title: 'Desenvolvimento de aplicativos web escaláveis com Next.js',
  excerpt:
    'Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.',
  createdAt: '2023-03-27T17:22:39.749Z',
  cover:
    'https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
