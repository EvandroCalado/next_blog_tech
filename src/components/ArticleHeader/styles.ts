import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
