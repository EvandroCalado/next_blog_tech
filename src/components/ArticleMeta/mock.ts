export default {
  createdAt: '2023-03-27T17:22:39.749Z',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
