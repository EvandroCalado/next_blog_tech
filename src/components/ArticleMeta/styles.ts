import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
