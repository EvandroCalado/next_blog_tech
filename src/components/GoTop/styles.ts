import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    position: fixed;
    bottom: 3rem;
    right: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    text-decoration: none;
    transition: all 300ms ease-in-out;

    &:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
