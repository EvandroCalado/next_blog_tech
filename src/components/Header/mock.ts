import { HeaderProps } from '.';

export default {
  blogName: 'Evandro Calado',
  blogDescription:
    'Lorem ipsum dolor sit amet consectetur adipisicing elit. Unde harum optio, eius quia quas molestias ipsum modi sed ullam exercitationem.',
  logo: 'https://res.cloudinary.com/dnhsdenji/image/upload/v1679924887/110628201_ae0753864b.jpg',
  showText: true,
} as HeaderProps;
