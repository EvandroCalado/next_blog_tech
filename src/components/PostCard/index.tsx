import Link from 'next/link';
import Heading from '../Heading';
import * as Styled from './styles';

export type PostCardProps = {
  id: string;
  title: string;
  slug: string;
  cover: string;
  excerpt: string;
};

const PostCard = ({ title, slug, cover, excerpt }: PostCardProps) => {
  return (
    <Styled.Wrapper>
      <Link href={`/post/${slug}`}>
        <Styled.Cover src={cover} alt={title} />
      </Link>
      <Heading as="h2" size="small">
        <Link href={`/post/${slug}`}>{title}</Link>
      </Heading>
      <Styled.Excerpt>{excerpt}</Styled.Excerpt>
    </Styled.Wrapper>
  );
};

export default PostCard;
