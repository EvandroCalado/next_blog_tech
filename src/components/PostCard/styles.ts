import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    ${Title} {
      margin: ${theme.spacings.small} 0;
    }
    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    &:hover a {
      color: ${theme.colors.secondary};
    }

    &:hover img {
      opacity: 0.8;
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
  aspect-ratio: 16/9;
  transition: opacity 300ms ease-in-out;
`;

export const Excerpt = styled.p``;
