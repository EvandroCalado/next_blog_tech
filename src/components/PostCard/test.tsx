import PostCard from '.';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<PostCard />', () => {
  it('should render a heading, cover and excerpt', () => {
    renderTheme(<PostCard {...mock} />);
    expect(
      screen.getByRole('heading', { name: mock.title }),
    ).toBeInTheDocument();
    expect(screen.getByRole('img', { name: mock.title })).toBeInTheDocument();
    expect(
      screen.getAllByRole('link', { name: mock.title })[0],
    ).toHaveAttribute('href', `/post/${mock.slug}`);
    expect(screen.getByText(mock.excerpt)).toBeInTheDocument();
  });

  it('should render match snapshot', () => {
    const { container } = renderTheme(<PostCard {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
