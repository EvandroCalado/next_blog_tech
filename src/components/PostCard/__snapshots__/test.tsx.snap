// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should render match snapshot 1`] = `
.c3 {
  color: #2a3840;
  line-height: 1.2;
  font-size: 2.4rem;
  font-weight: 900;
  text-transform: none;
  text-align: center;
}

.c0 .c2 {
  margin: 1.6rem 0;
}

.c0 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover a {
  color: #f24b4b;
}

.c0:hover img {
  opacity: 0.8;
}

.c1 {
  max-width: 100%;
  -webkit-transition: opacity 300ms ease-in-out;
  transition: opacity 300ms ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <a
      href="/post/desenvolvimento-de-aplicativos-web-escalaveis-com-next-js"
    >
      <img
        alt="Desenvolvimento de aplicativos web escaláveis com Next.js"
        class="c1"
        src="https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png"
      />
    </a>
    <h2
      class="c2 c3"
    >
      <a
        href="/post/desenvolvimento-de-aplicativos-web-escalaveis-com-next-js"
      >
        Desenvolvimento de aplicativos web escaláveis com Next.js
      </a>
    </h2>
    <p
      class=""
    >
      Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.
    </p>
  </div>
</div>
`;
